app/internal/page/app_test.go

```diff
@@ -333,6 +333,55 @@ func TestTestingStart(t *testing.T) {
 		assert.Equal(t, "/somewhere", resp.Header.Get("Location"))
 		mock.AssertExpectationsForObjects(t, sessionsStore, lpaStore)
 	})
+
+	t.Run("how attorneys act", func(t *testing.T) {
```

Comment:
@hawx - I'm pretty sure this test passes no matter what the props on the LPA are set to. I'd expect `AssertExpectationsForObjects()` to trigger a failure if `lpa.DecisionType` doesn't match but it passes anyway. Any idea what I'm missing? This applies to the other tests in this series too after I tried to break some of them.

Reply:
I think the problem is that the `Get` call will return `lpa` (which is a pointer, so the real code will mutate the value), then the `Put` mock will obviously have the right values because the code has just changed them (so your setting of `DecisionsType` will be overwritten).

If you inline `lpa` you'll see the test can be broken:

```go

lpaStore := &mockLpaStore{}
lpaStore.
	On("Get", ctx, mock.Anything).
	Return(&Lpa{}, nil)

lpaStore.
	On("Put", ctx, mock.Anything, &Lpa{DecisionsType: tc.DecisionsType, DecisionsDetails: "what"}). // e.g. for the failure
	Return(nil)
```

Reply:
Pointers again 😂 cheers for that, will update!

Reply:
Yep, I'd just recommend writing things out in full in the tests because mutating can introduce subtle issues like this